robot_modeling/DQ_SerialManipulatorMDH.m

```diff
@@ -0,0 +1,306 @@
+        % Prismatic joint
+        JOINT_PRISMATIC = 2;  % Deprecated
+    end
+    
+    methods (Access = protected)
+        function dq = mdh2dq(obj,q,ith)
+            %   MDQ2DQ(q, ith) calculates  the corresponding dual quaternion for
+            %   a given link's modified DH parameters
+            %
+            %   Usage: dq = mdh2dq(q,ith), where
+            %          q: joint value
+            %          ith: link number
+            %
+            %   Eq. (2.34) of Adorno, B. V. (2011). Two-arm Manipulation: From Manipulators
+            %   to Enhanced Human-Robot Collaboration [Contribution à la manipulation à deux bras : 
+            %   des manipulateurs à la collaboration homme-robot]. 
+            %   https://tel.archives-ouvertes.fr/tel-00641678/
+            
+            if nargin ~= 3
+                error('Wrong number of arguments. The parameters are joint value and the correspondent link')
+            end
+            
+            % Store half angles and displacements
+            half_theta = obj.theta(ith)/2.0;
+            d = obj.d(ith);
+            a = obj.a(ith);
+            half_alpha = obj.alpha(ith)/2.0;
+            
+            % Add the effect of the joint value
+            if obj.type(ith) == DQ_JointType.REVOLUTE
+                % If joint is revolute
+                half_theta = half_theta + (q/2.0);
+            else
+                % If joint is prismatic
+                d = d + q;
+            end
+            
+            % Pre-calculate cosines and sines
+            sine_of_half_theta = sin(half_theta);
+            cosine_of_half_theta = cos(half_theta);
+            sine_of_half_alpha = sin(half_alpha);
+            cosine_of_half_alpha = cos(half_alpha);
+            
+            d2 = d/2;
+            a2 = a/2;
+            h(1) = cosine_of_half_alpha*cosine_of_half_theta;
+            h(2) = sine_of_half_alpha*cosine_of_half_theta;
+            h(3) = -sine_of_half_alpha*sine_of_half_theta;
+            h(4) = cosine_of_half_alpha*sine_of_half_theta;
+            h(5) = -a2*h(2) - d2*h(4);
+            h(6) =  a2*h(1) - d2*-h(3);
+            h(7) = -a2*h(4) - d2*h(2);
+            h(8) = d2*h(1)  - a2*-h(3);
+            dq = DQ(h);
+        end 
+
+        function w = get_w(obj,ith) 
+        % This method returns the term 'w' related with the time derivative of 
+        % the unit dual quaternion pose using the Modified DH convention.
+        % See. eq (2.32) of 'Two-arm Manipulation: From Manipulators to Enhanced 
+        % Human-Robot Collaboration' by Bruno Adorno.
+        % Usage: w = get_w(ith), where
+        %          ith: link number
+
+            if obj.type(ith) == DQ_JointType.REVOLUTE            
+                w = -DQ.j*sin(obj.alpha(ith))+ DQ.k*cos(obj.alpha(ith))...
+                    -DQ.E*obj.a(ith)*(DQ.j*cos(obj.alpha(ith)) + DQ.k*sin(obj.alpha(ith)));
+            else % if joint is PRISMATIC          
```

Comment:
Hi @juanjqo,

I double-checked the calculations for the prismatic joint and they are correct. 🥳

Reply:
@bvadorno

Nice! :bowtie::bowtie::bowtie::santa::santa::santa::santa: